pydicom/pixel_data_handlers/gdcm_handler.py

```diff
@@ -122,9 +122,15 @@ def get_pixeldata(dicom_dataset):
     n_bytes = (dicom_dataset.Rows *
                dicom_dataset.Columns *
                numpy.dtype(numpy_dtype).itemsize)
-
+    try:
+        n_bytes *= dicom_dataset.NumberOfFrames
+    except Exception:
+        pass
+    try:
+        n_bytes *= dicom_dataset.SamplesPerPixel
+    except Exception:
+        pass
     if len(pixel_bytearray) > n_bytes:
```

Comment:
Maybe use instead:
``` python
    n_bytes *= dicom_dataset.get('NumberOfFrames', 1)
    n_bytes *= dicom_dataset.get('SamplesPerPixel', 1)

Reply:
I like this suggestion too -- much easier to understand what is going on.  In any case, I'd prefer not having a bare Exception, it should catch the specific error.

Reply:
I wish python had a 
```
    try_each:
        Statement1
        Statement2
    except:
        HandleExceptionStatements
```

that would be the same as the try...except, try...except that I end up with...

Reply:
Did you change this occurrence too, or only the one below?